polkadot/zombienet-sdk-tests/tests/smoke/coretime_revenue.rs

```diff
@@ -0,0 +1,514 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Binaries for this test should be built with `fast-runtime` feature enabled:
+//! `cargo build -r -F fast-runtime -p polkadot-parachain-bin && \`
+//! `cargo build -r -F fast-runtime --bin polkadot --bin polkadot-execute-worker --bin
+//! polkadot-prepare-worker`
+//!
+//! Running with normal runtimes is possible but would take ages. Running fast relay runtime with
+//! normal parachain runtime WILL mess things up.
+
+use anyhow::anyhow;
+#[subxt::subxt(runtime_metadata_path = "metadata-files/rococo-local.scale")]
+pub mod rococo {}
+
+#[subxt::subxt(runtime_metadata_path = "metadata-files/coretime-rococo-local.scale")]
+mod coretime_rococo {}
+
+use rococo::runtime_types::{
+	staging_xcm::v4::{
+		asset::{Asset, AssetId, Assets, Fungibility},
+		junction::Junction,
+		junctions::Junctions,
+		location::Location,
+	},
+	xcm::{VersionedAssets, VersionedLocation},
+};
+use serde_json::json;
+use std::{fmt::Display, sync::Arc};
+use subxt::{events::StaticEvent, utils::AccountId32, OnlineClient, PolkadotConfig};
+use subxt_signer::sr25519::dev;
+use tokio::sync::RwLock;
+use zombienet_sdk::NetworkConfigBuilder;
+
+use coretime_rococo::{
+	self as coretime_api,
+	broker::events as broker_events,
+	runtime_types::{
+		pallet_broker::types::{ConfigRecord as BrokerConfigRecord, Finality as BrokerFinality},
+		sp_arithmetic::per_things::Perbill,
+	},
+};
+
+use rococo::{self as rococo_api, runtime_types::polkadot_parachain_primitives::primitives};
+
+type CoretimeRuntimeCall = coretime_api::runtime_types::coretime_rococo_runtime::RuntimeCall;
+type CoretimeUtilityCall = coretime_api::runtime_types::pallet_utility::pallet::Call;
+type CoretimeBrokerCall = coretime_api::runtime_types::pallet_broker::pallet::Call;
+
+// On-demand coretime base fee (set at the genesis)
+//
+// NB: This fee MUST always be higher than RC's existential deposit value. Otherwise, when
+//     some insta coretime is bought for the base fee value, it gets transferred from the buyer's
```

Comment:
We settled on on-demand coretime.

Reply:
Just to be sure: This is only a restriction for this test - right?

Reply:
Shouldn't  we not just not make the pot empty - ever?

Reply:
It's not even a restriction anymore, indeed. This comment was written before we added a pallet provider for the pot account, so now it's free to fall under the ED, and the account won't be deleted, so this comment is outdated.

Reply:
If it's outdated can we just remove it?